README.md

```diff
@@ -109,6 +111,11 @@ function MyObject () {
 }
 ```
 
+Also note how in the above examples, the code, that consumes an istance of `MyObject`,
+reset the state to initial condition, just before storing it in the cache.
+That's needed so that every subsequent request for an instance from the cache,
```

Comment:
@mcollina could you think of any use case in which it's more convenient to maintain the state?